### ISSUE TITLE
Not able to set FEC parameter by default from hwsku.json

### ISSUE THREAD

**Issue by dmytroxshevchuk** [Azure/sonic-buildimage#6495]
**Description**
After deploy new platform.json, hwsku.json files, don`t have possibility to set FEC parameter for port as it was in port_config.ini.
For example from port_config.ini:
```
name          lanes                 alias       index    speed    autoneg   fec
Ethernet0       0,1                   Ethernet0       1    25000   0         fc
Ethernet2       2,3                   Ethernet2       1    25000   0         fc
```
In hwsku.json we don`t have opportunity to set FEC parameter.

**Steps to reproduce the issue:**
1. deploy platform.json, hwsku.json
2. generate config_db and install it
3. reboot sonic

**Describe the results you received:**
In generated config_db.json file we don`t have FEC parameter:
```
    "PORT": {
        "Ethernet0": {
            "alias": "Ethernet0",
            "lanes": "0,1,2,3",
            "speed": "100000",
            "index": "1",
            "admin_status": "up",
            "mtu": "9100"
        },
```
Also it not exist in config_db, check it using command ```show interface status```.
```
  Interface        Lanes    Speed    MTU    FEC       Alias    Vlan    Oper    Admin    Type    Asym PFC
-----------  -----------  -------  -----  -----  ----------  ------  ------  -------  ------  ----------
  Ethernet0      0,1,2,3     100G   9100     N/A   Ethernet0  routed    down     down     N/A         N/A
  Ethernet8    8,9,10,11     100G   9100     N/A   Ethernet8  routed    down       up     N/A         N/A
```
As we can see FEC is N/A.

**Describe the results you expected:**
I expect that, must be opportunity to set FEC parameter into hwsku.json file.
Expect config_db.json: 
```
    "PORT": {
        "Ethernet0": {
            "alias": "Ethernet0",
            "lanes": "0,1,2,3",
            "speed": "100000",
            "index": "1",
            "admin_status": "up",
            "fec": "rs",
            "mtu": "9100"
        },
```
Expect output after ```show interface status```:
```
  Interface        Lanes    Speed    MTU    FEC       Alias    Vlan    Oper    Admin    Type    Asym PFC
-----------  -----------  -------  -----  -----  ----------  ------  ------  -------  ------  ----------
  Ethernet0      0,1,2,3     100G   9100     rs   Ethernet0  routed    down     down     N/A         N/A
  Ethernet8    8,9,10,11     100G   9100     rs   Ethernet8  routed    down       up     N/A         N/A
```

**Additional information you deem important (e.g. issue happens only occasionally):**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

**Comment by dmytroxshevchuk** [comment]
@zhenggen-xu @daall @praveen-li , please take a look

**Comment by zhenggen-xu** [comment]
@dmytroxshevchuk  FC configuration should come from config_db.json, if not configured, then it is leave to orchagent or SAI for the default.  If you want to define the default for each hwsku.json,  we may have to add this capability in, so that it could translate to PORT table during the initialization.

**Comment by dmytroxshevchuk** [comment]
> @dmytroxshevchuk FC configuration should come from config_db.json, if not configured, then it is leave to orchagent or SAI for the default. If you want to define the default for each hwsku.json, we may have to add this capability in, so that it could translate to PORT table during the initialization.

@zhenggen-xu , we need define the default for each hwsku.json. Please take a look this PR: https://github.com/Azure/sonic-buildimage/pull/6155. This PR parse additional fields from hwsku.json, so we can define FEC parameter by default and it will be translate to PORT table during the initialization.

**Comment by akokhan** [comment]
As per DPB HLD:
"**platform.json** and **hwsku.json** files will deprecate the old **port_config.ini** file in the current SONiC design"

Looks like without fixing this issue, we can not deprecate **port_config.ini** which contains both autoneg & fec.

**Comment by zhenggen-xu** [comment]
Yes, I agree we should fix it.  I left some comments in  https://github.com/Azure/sonic-buildimage/pull/6155 .

**Comment by lguohan** [comment]
fixed by #6155
